src/jobsautoreport/report.py

```diff
@@ -1,31 +1,188 @@
-        successful_jobs_list = self._querier.query_successful_jobs(from_date, to_date)
-        unsuccessful_jobs_list = self._querier.query_successful_jobs(from_date, to_date)
-        if len(successful_jobs_list) == 0:
-            return "0"
-        return f"{(len(successful_jobs_list) / (len(unsuccessful_jobs_list) + len(successful_jobs_list))) * 100}%"
+    @staticmethod
+    def _get_number_of_rehearsal_jobs(jobs: list[JobDetails]):
+        return len(jobs)
+
+    @staticmethod
+    def _get_job_triggers_count(job_name: str, jobs: list[JobDetails]) -> int:
+        return sum(1 for job in jobs if job.name == job_name)
 
-    def get_number_of_jobs_triggered(
-        self, from_date: datetime, to_date: datetime
+    @staticmethod
+    def _get_job_success_rate(job_name: str, jobs: list[JobDetails]) -> float:
+        total_times = len([job for job in jobs if job.name == job_name])
+        successful_times = len(
+            [
+                job
+                for job in jobs
+                if job.name == job_name and job.state == JobState.SUCCESS.value
+            ]
+        )
+        return (
+            0
+            if total_times == 0
+            else float(round((successful_times / total_times) * 100, 2))
+        )
+
+    @staticmethod
+    def _get_number_of_jobs_by_state(
+        jobs: list[JobDetails], job_state: JobState
     ) -> int:
-        jobs_list = self._querier.query_number_of_jobs_triggered(from_date, to_date)
-        return len(jobs_list)
-
-    def get_average_duration_of_jobs(
-        self, from_date: datetime, to_date: datetime
-    ) -> Optional[timedelta]:
-        jobs_list = self._querier.query_number_of_jobs_triggered(from_date, to_date)
-        if len(jobs_list) == 0:
-            return None
-        avg = int(sum([job.duration for job in jobs_list]) / len(jobs_list))
-        return timedelta(seconds=avg)
+        return len([job for job in jobs if job.state == job_state.value])
+
+    @classmethod
+    def _get_success_rate(cls, jobs: list[JobDetails]) -> float:
+        total_jobs_number = len(jobs)
+        successful_jobs_number = cls._get_number_of_jobs_by_state(
+            jobs, JobState.SUCCESS
+        )
+        return (
+            0
+            if total_jobs_number == 0
+            else float(round((successful_jobs_number / total_jobs_number) * 100, 2))
+        )
+
+    @staticmethod
+    def _get_top_n_jobs(
+        jobs: list[JobDetails],
+        computation_func: Callable[[str, list[JobDetails]], Any],
+        n: int,
+        top_successful: bool = False,
+    ):
+        distinct_jobs = {job.name for job in jobs}
+        res = [
+            (job_name, computation_func(job_name, jobs)) for job_name in distinct_jobs
+        ]
+        res = sorted(res, key=lambda job: (job[1], job[0]), reverse=top_successful)
+        return res[0 : min(len(res), n)]
+
+    @classmethod
+    def _filter_out_non_e2e_nor_subsystem_jobs(
```

Comment:
```suggestion
    def _get_e2e_or_subsystem_jobs(
```
I think it's easier to understand